### DISCUSSION TITLE
StrawberryShake CICD integration

### DISCUSSION THREAD

**Discussion by bieri** [ChilliCream/hotchocolate#5491] [General]
I could use some help. I would like to generate the StrawberryShake Client from my CICD pipeline, but the API that I am connecting to requires MFA to create the authentication tokens required to generate the client.  I do however have access to the raw graphql schema. Is there a way to generate the clients with the raw schema and skipping the attempt to reach out to the remote server?
Thanks

**Comment by tobias-tengler** [comment]
You normally shouldn't have to do any remote calls in your CI as the schema file should be part of your repository. If you have access to the raw schema, you can just swap out the schema.graphql file with your schema file. The generation of the client is triggered by building the project and based on the schema file. You don't have to run the `dotnet graphql` stuff to generate the client.

**Reply by bieri** [reply]
Thanks for explaining to me, I just needed to recreate the csproj file to get it all working.
